src/librustc_typeck/diagnostics.rs

```diff
@@ -4567,6 +4567,52 @@ i_am_a_function();
 ```
 "##,
 
+E0619: r##"
+A not (yet) known type was used.
+
+Erroneous code example:
+
+```compile_fail,E0619
+let x;
```

Comment:
This example is simple, but it does conflate an *uninitialized value* with one where we do not know the type. An alternative might be something like this:

```rust
let mut x = vec![];
match x.pop() {
    Some(v) => {
        // Here, the type of `v` is not (yet) known, so we
        // cannot resolve this method call:
        v.to_uppercase();
    }
    None => { }
}
```

This could be fixed by annotating `x` with a type, like `x: Vec<String>`. To make it more realistic, we'd need a `loop` or something else.

Another example that I encounter a lot:

```rust
let vec1 = vec![1, 2, 3];
let vec2 = vec1.iter().map(|x| x * 2).collect();
println!("{}", vec2.len()); // error
```

with the fix:

```rust
let vec1 = vec![1, 2, 3];
let vec2: Vec<_>  = vec1.iter().map(|x| x * 2).collect();
println!("{}", vec2.len()); // error
```

the problem with this one, of course, is that it references the iterator API. But we could probably explain it just by saying that "because `collect()` can create many kinds of collections, the type is typically inferred from context." or something like that.